content/trial_getting_started/examples/kubernetes/_index.md

```diff
@@ -0,0 +1,99 @@
+alwaysopen = false
++++
+
+This example demonstrates defining an environment independent Kubernetes workload and deploying it to development and production clusters.  The example runs on Amazon Web Services (AWS) and so requires and account and API credentials.  It available on [github](https://github.com/cloudify-community/eaas-example).
+
+The infrastructure deployment consists of [`ServiceComponent`s](../../../working_with/service_composition) representing:
+
+ * network
+ * database
+ * object store
+ * kubernetes
+
+The implementation of these components is based on the environment type, which is supplied
+as an input to the deployment.  For example, if a development environment is selected, <a href="http://minio.io">Minio</a> is used as the object store and <a href="https://minikube.sigs.k8s.io/docs/">Minikube</a> as the Kubernetes provider.  If a production environment is selected, <a href="https://aws.amazon.com/s3">S3</a> and <a href="https://aws.amazon.com/eks">EKS</a> is used for each, respectively.
+
+## Concepts
+
+### Configuration Indirection
+
+Multi-Kubernetes Cluster orchestration is achieved using a combination of the plugins mentioned above, and the design technique introduced in the [Environment as a Service](../eaas) example.  The technique uses a blueprint node as the source of cluster configuration.  This configuration is populated by a script which takes the kind of environment desired as an input.  The particulars of the configuration are then looked up and stored in runtime properties on the node.  Other nodes then use this configuration via the `get_attribute` intrinsic function.
+
+### Service Composition
+
+The configuration indirection helps with mapping simple identifiers (like 'dev-small') to complex configuration details (like image names/id, flavors, etc..).  This is not sufficient to completely abstract away the different Kubernetes environments required.  To do this requires [service composition](../../../working_with/service_composition).  Service composition allows blueprint nodes to represent entire blueprints themselves, effectively nesting blueprints and enabling a building block approach.  The main requirement to be able to use components in the way required by this example is that components of a similar kind (for example, blueprints that represent different kinds of database), all have a consistent interface.  In Cloudify DSL, this interface is provided by the `capabilitites` section in the blueprint.
+
+Looking at the [multi-Kubernetes cluster example](https://github.com/cloudify-community/eaas-example), consider the object storage options `minio` and `S3`.  Both of these are represented by a blueprint that exposes a single `capability`: `bucket_url`.  Because of this standard "interface", the blueprints can be substituted for each other at deploy time.  You will find the same pattern for other elements: minikube/EKS (`endpoint`), psql/RDS (`host`, `master_username`, `master_password`), and so on.
+
```

Comment:
Can we provide links to the exact files, and also provide here an inline example of what is being discussed?

Reply:
Yeah, I was consciously avoiding that.  But I can dig into Python I suppose

Reply:
Never mind.  It appears it was rewritten to not use Python.  Will rewrite the section.